src/metabase/driver/duckdb.clj

```diff
@@ -131,6 +131,26 @@
    db-or-id-or-spec
    options
    (fn [^Connection conn]
+     (when (not (sql-jdbc.execute/recursive-connection?))
+       (when-let [init-sql (-> db-or-id-or-spec :details :init_sql)]
+         (when (seq (str/trim init-sql))
+           (with-open [stmt (.createStatement conn)]
+             ;; Ensure init sql only executes once per DuckDB instance using a DuckDB variable
+             (let [rs (.executeQuery stmt "SELECT getvariable('initialized');")]
+               (when (.next rs)
+                 (let [initialized? (.getBoolean rs 1)]
+                   (log/infof "DuckDB init SQL executed: %s" initialized?)
+                   (when-not initialized?
+                     (log/infof "DuckDB init SQL has not been executed, executing now...")
+                     (try
+                       (log/infof "Executing DuckDB init SQL: %s" init-sql)
```

Comment:
init sql for ducklake is fairly likely to contain secret setting. It would be better not to log